File: server/datasets/python_source-code/Multiple-G3/renamed_44_dsu.py
class DSU_8:
    def __init__(self, n_8):
        self.p_8 = list(range(n_8))
        self.rk_8 = [0] * n_8

    def find_8(self, x_8):
        if self.p_8[x_8] != x_8:
            self.p_8[x_8] = self.find_8(self.p_8[x_8])
        return self.p_8[x_8]

    def union_8(self, x_8_a, x_8_b):
        ra_8 = self.find_8(x_8_a)
        rb_8 = self.find_8(x_8_b)
        if ra_8 != rb_8:
            if self.rk_8[ra_8] < self.rk_8[rb_8]:
                ra_8, rb_8 = rb_8, ra_8
            self.p_8[rb_8] = ra_8
            if self.rk_8[ra_8] == self.rk_8[rb_8]:
                self.rk_8[ra_8] += 1
            return True
        return False

File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_g3/renamed_44_dsu.java
public class DSU_8 {
    public static void main(String[] args) {
        DSU_8 dsu = new DSU_8(10);
        dsu.u_8(1, 2);
    }
    int[] p_8;
    int[] r_8;
    public DSU_8(int tot_8) {
        p_8 = new int[tot_8];
        r_8 = new int[tot_8];
        for (int idx = 0; idx < tot_8; idx++) {
            p_8[idx] = idx;
            r_8[idx] = 0;
        }
    }
    public int f_8(int x_8) {
        if (p_8[x_8] != x_8) {
            p_8[x_8] = f_8(p_8[x_8]);
        }
        return p_8[x_8];
    }
    public void u_8(int a_8, int b_8) {
        int ra_8 = f_8(a_8);
        int rb_8 = f_8(b_8);
        if (ra_8 != rb_8) {
            if (r_8[ra_8] < r_8[rb_8]) {
                p_8[ra_8] = rb_8;
            } else if (r_8[ra_8] > r_8[rb_8]) {
                p_8[rb_8] = ra_8;
            } else {
                p_8[rb_8] = ra_8;
                r_8[ra_8]++;
            }
        }
    }
}